### PR TITLE
Fix occasional failure in UI listener.

### PR DESCRIPTION
**Pull request by pwendell**
If a task fails before the metrics are initialized, it remains possible
that the metrics field will be `None`. This patch accounts for that possbility
by keeping metrics as an `Option` at all times.

### CODE REVIEW

File: core/src/main/scala/spark/ui/jobs/StagePage.scala [hunk 1]
```diff
@@ -52,7 +52,7 @@ private[spark] class StagePage(parent: JobProgressUI) {
       }
       else {
         val serviceTimes = validTasks.map{case (info, metrics, exception) =>
-          metrics.executorRunTime.toDouble}
+          metrics.get.executorRunTime.toDouble}
```
Comment (ryanlecompte) [inline core/src/main/scala/spark/ui/jobs/StagePage.scala:55]: Is it safe to always call `metrics.get` in these cases? Could it ever be `None` ?

Comment (pwendell) [inline core/src/main/scala/spark/ui/jobs/StagePage.scala:55]: Yes, all `validTasks` have metrics, that's the definition of `validTasks`...

Comment (ryanlecompte) [inline core/src/main/scala/spark/ui/jobs/StagePage.scala:55]: Sweet!

**Comment by mateiz** [comment]
I've tested this manually and merged it in. Thanks Patrick!

**Pull request merged**